cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/CatalogUtil.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.catalog.migration.cli;
+
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.aws.dynamodb.DynamoDbCatalog;
+import org.apache.iceberg.aws.glue.GlueCatalog;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.dell.ecs.EcsCatalog;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.apache.iceberg.jdbc.JdbcCatalog;
+import org.apache.iceberg.nessie.NessieCatalog;
+import org.apache.iceberg.rest.RESTCatalog;
+
+public final class CatalogUtil {
+
+  private CatalogUtil() {}
+
+  public enum CatalogType {
+    CUSTOM,
+    DYNAMODB,
+    ECS,
+    GLUE,
+    HADOOP,
+    HIVE,
+    JDBC,
+    NESSIE,
+    REST
+  }
+
+  static Catalog buildCatalog(
+      Map<String, String> catalogProperties,
+      CatalogType catalogType,
+      String customCatalogImpl,
+      Map<String, String> hadoopConf) {
+    Preconditions.checkArgument(catalogProperties != null, "catalog properties is null");
+    Preconditions.checkArgument(catalogType != null, "catalog type is null");
+    Configuration catalogConf = new Configuration();
+    if (hadoopConf != null) {
+      hadoopConf.forEach(catalogConf::set);
+    }
+    return org.apache.iceberg.CatalogUtil.loadCatalog(
```

Comment:
If this one fails, the user cannot distinguish whether the source or target catalog init failed.

```
java.lang.IllegalArgumentException: Cannot initialize HadoopCatalog because warehousePath must not be null or empty
	at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at org.apache.iceberg.hadoop.HadoopCatalog.initialize(HadoopCatalog.java:106)
```

Reply:
I fixed it previously by passing the prefix `SOURCE_CATALOG_` and `TARGET_CATALOG_` In the caller `SourceCatalogOptions` and `TargetCatalogOptions`.  So that this user can distinguish which catalog has failed.